enhancements/testing/openshift-tests-extension.md

```diff
@@ -0,0 +1,1202 @@
+    * [Test Extension Binaries](#test-extension-binaries)
+      * [Binary Discovery](#binary-discovery)
+        * [OpenShift Payload Extension Binaries](#openshift-payload-extension-binaries)
+        * [Non-Payload Extension Binaries](#non-payload-extension-binaries)
+      * [Binary Format](#binary-format)
+      * [Binary Extraction](#binary-extraction)
+      * [Extension Interface](#extension-interface)
+        * [Info - Extension Metadata](#info---extension-metadata)
+        * [List Tests - Extension Test Listing](#list-tests---extension-test-listing)
+        * [List Monitors - List which monitoring tests are available](#list-monitors---list-which-monitoring-tests-are-available)
+        * [Run-Test - Running Extension Tests](#run-test---running-extension-tests)
+        * [Run-Suite - Running Tests in Local Suites](#run-suite---running-tests-in-local-suites)
+        * [Run-Monitor - Monitoring Cluster during Test Run](#run-monitor---monitoring-cluster-during-test-run)
+        * [Config - Component Configuration Testing](#config---component-configuration-testing)
+      * [Update - Metadata Validation](#update---metadata-validation)
+      * [Extension Implementation](#extension-implementation)
+      * [Test Result Aggregation](#test-result-aggregation)
+    * [Risks and Mitigations](#risks-and-mitigations)
+      * [Binary Incompatibility](#binary-incompatibility)
+        * [CPU Architecture](#cpu-architecture)
+      * [Runtime Size / Speed](#runtime-size--speed)
+      * [Image Size](#image-size)
+      * [Poor Extension Implementation](#poor-extension-implementation)
+    * [Version Skew Strategy](#version-skew-strategy)
+  * [Alternatives](#alternatives)
+<!-- TOC -->
+
+# OpenShift Tests Extensions
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Today all conformance tests for OpenShift are in the monolithic [openshift-tests]
+(https://www.github.com/openshift/origin) binary. These tests run on all the possible
+flavors of OpenShift (more than 50) and contribute signal to tools like the
+Release Controller and Component Readiness. However, the high barrier to entry in
+openshift-tests means that teams are either not writing tests, or are using their
+own homegrown solutions that ran in only very narrow configurations.
+
+This enhancement proposes a framework to allow allows external repositories to
+contribute tests to openshift-tests' suites with extension binaries, keeping tests
+colocated with the features they are testing. It defines a standardized interface
+for test discovery, execution, and result aggregation, allowing decentralized
+contributions while maintaining centralized orchestration.

```

Comment:
> allowing decentralized contributions while maintaining centralized orchestration.

Is there any impact in how the new tests are added to the conformance suite, such as `openshift/conformance`?

Reply:
Common suites can be extended by adding tests to component repositories and citing those suites as parents; e.g. https://github.com/openshift-eng/openshift-tests-extension/blob/6cbf1d70a4500e2483275aff3dc6a74afe9c6365/cmd/example-tests/main.go#L27-L31 . Once the binary is registered, openshift-tests will run the inherited tests as part of the associated conformance suite.